sklearn/ensemble/forest.py

```diff
@@ -436,8 +473,10 @@ def _set_oob_score(self, X, y):
                        for k in range(self.n_outputs_)]
 
         for estimator in self.estimators_:
+            n_bootstrap_samples = _get_n_bootstrap_samples(
+                n_samples, self.max_samples)
             unsampled_indices = _generate_unsampled_indices(
-                estimator.random_state, n_samples)
+                estimator.random_state, n_samples, n_bootstrap_samples)
```

Comment:
@jnothman here we will test on the left out samples. Since we used `max_samples` it means that we will use a lot of samples for the OOB. Is it something that we want?

If this is the case, we should add a test to make sure that we have the right complement in case `max_samples` is not None.

Reply:
Is there a problem with having a large OOB sample for test? Testing with trees isn't fast, but is a lot faster than training...?

Reply:
> Testing with trees isn't fast, but is a lot faster than training...?

True. Should not be an issue then

Reply:
If the sample is much smaller than the dataset I suppose it may be an issue. Maybe we should make a rule that the oob sample is constrained to be no larger than the training set... But that may confuse users trying different values for this parameter

Reply:
> If the sample is much smaller than the dataset

I can see this situation easily arising, e.g, your dataset is 10<sup>6</sup> examples and you want to to fit with say, `max_samples=1000` and `n_estimators=1000`.